From 264e573f22aa7c418e4b852bc93b59f9864ec687 Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Sat, 9 Dec 2017 11:38:00 -0600
Subject: [PATCH] overlay: Disable SystemUI anti-falsing on lockscreen

 * The anti-falsing implementation from HumanInteractionClassifier
    regularly prevents easy swipe to unlock or to pattern / pin
    on the keyguard lockscreen, requiring multiple attempts
    until accepted due to a hardcoded evaluation (5.0f)
    while normal usage shows better results without it

 * Another solved situation is remote device access like
    Vysor or TeamViewer where the device is almost impossible
    to swipe properly from a computer client

Change-Id: I1caf603bc6a3bf7794af724b8e48a9ebb96da971
---
 .../common/frameworks/base/packages/SystemUI/res/values/config.xml   | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml b/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml
index 840a615..162abaa 100644
--- a/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml
+++ b/overlay/common/frameworks/base/packages/SystemUI/res/values/config.xml
@@ -26,4 +26,9 @@
     <bool name="config_show4GForLTE">false</bool>
 
     <bool name="quick_settings_show_full_alarm">true</bool>
+
+    <!-- If true, enable the advance anti-falsing classifier on the lockscreen. On some devices it
+         does not work well, particularly with noisy touchscreens. Note that disabling it may
+         increase the rate of unintentional unlocks. -->
+    <bool name="config_lockscreenAntiFalsingClassifierEnabled">false</bool>
 </resources>
-- 
2.7.4

